### ISSUE TITLE
Removal of properties that shouldn't be

### ISSUE THREAD

**Issue by cferdinandi** [cferdinandi/reef#33]
Currently, dynamic form properties that are not defined in a template (like `value` or `checked` are removed if the state updates). 

Look in to how to effectively "ignore" properties when diffing if they're not defined in the template.

```js
var app = new Reef('#app', {
	data: {
		submitTxt: 'Submit'
	},
	template: function (props) {
		return `
			<form>
				<label for="username">Username</label>
				<input type="text" id="username">

				<label for="password">Password</label>
				<input type="password" id="password">

				<button>${props.submitTxt}</button>
			</form>`;
	}
});

app.render();

document.addEventListener('submit', function(event) {
	event.preventDefault();
	app.setData({submitTxt: 'Submitting...'});
});
```

In this example, the fields would get wiped out when the form is in submitting state...

For my own reference: search [this page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#htmlattrdefdisabled) for `if present` to find out which fields this applies to.

**Comment by cferdinandi** [comment]
I may need to switch to ignoring certain properties (`checked`, `selected`, etc.), and force the use of the `default*` versions of those properties instead. This is what React does.

**Comment by cferdinandi** [comment]
Fixed with v4.1.14
